File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-0.java
class Solution {
    public int search(int[] nums, int target) {

        int l = 0;
        int r = nums.length - 1;

        while(l<=r){
            int m = (l+r)/2;

            if(nums[m] == target){
                return m;
            }

            // left portion
            if(nums[m] >= nums[l]){

                if(target < nums[l] || target > nums[m]){
                    l = m +1;
                }
                else
                {
                    r = m -1;
                }
            }

            //right portion
            else
            {
                if(target > nums[r]){
                    r = m -1;
                }else if (target < nums[m])
                { 
                    r = m -1;
                }
                else
                {
                    l =m +1 ;
                }
            }

        }

        return -1;
}
}

File: Data Structures & Algorithms/balanced-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def isBalanced(self, root: Optional[TreeNode]) -> bool:
        balanced = True
        def dfs(root):
            nonlocal balanced
            if not root:
                return 0;
            
            root.left = dfs(root.left)
            root.right = dfs(root.right)
            
            if abs(root.left - root.right) > 1:
                balanced = False
            
            return 1 + max(root.left,root.right)
        dfs(root)
        return balanced


        